### DISCUSSION TITLE
Account setup IBAN - BIC

### DISCUSSION THREAD

**Discussion by EricVanCaenenberghe** [firefly-iii/firefly-iii#6741] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

At account set-up, if you put IBAN-number and forget the BIC-code then it is impossible to change the account afterwards (whatever field: account name, starting balance, or even adding the missing BIC-code).
Given error: IBAN-number already exist.
The problem does not occur when IBAN and BIC are present from the start.

### Expected behaviour

Option 1: IBAN and BIC are a pair and have to be introduced together as a mandatory pair
option 2: BIC is not mandatory even when IBAN is present but then when changing anything in the account setup should not complain about an already existing IBAN-number.

### Debug information

.

### Steps to reproduce

1. set-up an account with IBAN and without BIC
2. change the account by adding the BIC
3. error appears

**Comment by JC5** [comment]
I can't seem to replicate this on the demo site. Are you sure you don't have an expense or a revenue account with the same IBAN?

**Comment by EricVanCaenenberghe** [comment]
Got it. Imported transactions from Bank 1, one of the transactions was a transfer for which the account wasn't setup. So I created the account without any problems up to the moment I wanted to change something. 
My story was incorrect but still a little issue. I know what to do as workaround .
